### PR TITLE
Replace Update method with field update methods

### PR DESCRIPTION
**Pull request by jakewan**
This update is to help support the preferred Terraform pattern of updating resource attributes individually rather than several at once.

Includes improved unit tests.

Tests can be run locally, or in a container (see README.md).

When merged into `master`, I'll tag it as v2.0.0, since this breaks the previous public API.

### CODE REVIEW

File: itm/dns_apps.go [hunk 1]
```diff
@@ -107,16 +121,13 @@ func (s *dnsAppsServiceImpl) Update(id int, opts *DNSAppOpts, publish bool) (*DN
...
-		log.Printf("Error issuing put request from DNSAppsServiceImpl.Update: %v", err)
+		log.Printf("[WARN] Error issuing put request from DNSAppsServiceImpl.Update: %v", err)
 		return nil, err
 	}
 	if 200 != resp.StatusCode {
-		return nil, &UnexpectedHTTPStatusError{
-			Expected: 200,
-			Got:      resp.StatusCode,
-		}
+		return nil, fmt.Errorf("Unexpected HTTP response. HTTP status code: %d. Body: %s", resp.StatusCode, resp.Body)
```
Comment (joshcody) [inline itm/dns_apps.go:130]: Does this line need an error level like the other errors? (like line 106?)

File: itm/dns_apps.go [hunk 2]
```diff
@@ -130,9 +141,7 @@ func (s *dnsAppsServiceImpl) Get(id int) (*DNSApp, error) {
 		return nil, err
 	}
 	if 200 != resp.StatusCode {
-		return nil, &UnexpectedHTTPStatusError{
-			Expected: 200,
-			Got:      resp.StatusCode}
+		return nil, fmt.Errorf("Unexpected HTTP response. HTTP status code: %d. Body: %s", resp.StatusCode, resp.Body)
```
Comment (joshcody) [inline itm/dns_apps.go:144]: Does this need an error level?

File: itm/dns_apps.go [hunk 3]
```diff
@@ -141,12 +150,9 @@ func (s *dnsAppsServiceImpl) Get(id int) (*DNSApp, error) {
 func (s *dnsAppsServiceImpl) Delete(id int) error {
 	resp, err := s.client.delete(getDNSAppPath(id))
 	if 204 != resp.StatusCode {
-		return &UnexpectedHTTPStatusError{
-			Expected: 204,
-			Got:      resp.StatusCode,
-		}
+		return fmt.Errorf("Unexpected HTTP response. HTTP status code: %d. Error: %s", resp.StatusCode, err)
```
Comment (joshcody) [inline itm/dns_apps.go:153]: Missing error level?

**Review by joshcody — COMMENTED** [review]
Aside from a few consistency comments, the overall PR looks good.

File: itm/dns_apps.go (hunk 1 above)
Comment (jakewan) [reply itm/dns_apps.go:130]: Good eye! Actually it's the other way around. The convention is to include the log level in calls to `log.Printf` and `log.Println`, not in the strings used to create error objects, so I will it on line 106.

File: itm/dns_apps.go [hunk 4]
```diff
@@ -76,28 +97,21 @@ func (s *dnsAppsServiceImpl) Create(opts *DNSAppOpts, publish bool) (*DNSApp, er
...
-		log.Printf("Error issuing post request from DNSAppsServiceImpl.Create: %v", err)
+		log.Printf("[WARN] Error issuing post request from DNSAppsServiceImpl.Create: %v", err)
 		return nil, err
 	}
 	if 201 != resp.StatusCode {
-		return nil, &UnexpectedHTTPStatusError{
-			Expected: 201,
-			Got:      resp.StatusCode,
-		}
+		return nil, fmt.Errorf("Unexpected HTTP response. HTTP status code: %d. Body: %s", resp.StatusCode, resp.Body)
```
Comment (jakewan) [inline itm/dns_apps.go:106]: Removed log level for consistency (log level goes on calls to `log.Printf` and `log.Println`).

**Pull request closed without merging**